enhancements/microshift/microshift-router-configuration.md

```diff
@@ -0,0 +1,437 @@
+---
+title: microshift-router-configuration
+authors:
+  - "@pacevedom"
+reviewers:
+  - "@eslutsky"
+  - "@copejon"
+  - "@ggiguash"
+  - "@pmtk"
+  - "@pliurh"
+  - "@jerpeter1"
+  - "@Miciah"
+approvers:
+  - "@dhellmann"
+api-approvers:
+  - None
+creation-date: 2024-01-08
+last-updated: 2024-01-24
+tracking-link:
+  - https://issues.redhat.com/browse/USHIFT-2089
+---
+
+# MicroShift router default configuration options
+
+## Summary
+MicroShift's default router is created as part of the platform, but does not
+allow configuring any of its specific parameters. For example, you can not
+disable the router or change its listening ports.
+
+In order to allow these operations and many more, a set of configuration options
+is proposed.
+
+## Motivation
+MicroShift's default ingress router comes as part of the product and is always
+deployed. As part of the start procedure, MicroShift will configure the router
+with fixed parameters, such as being always enabled, and tied to ports 80 and
+443. None of these are configurable for the user, and they also tie to how the
+router is exposed in the cluster.
+
+In order to allow more flexibility, the following configuration parameters shall
+be added:
+- Enable/Disable. Some use cases may be egress only, meaning they do not need
+  an ingress controller. In this case the router pod should not be scheduled
+  to save resources. There should be no rules associated to firewalld or
+  iptables either for an improved security posture.
+- Listening ports. Allow configuring ports other than 80 and 443 improves
```

Comment:
Since this is not something OpenShift does, we want to go into more detail about why it is necessary for MicroShift. This bullet point could talk about MicroShift not owning the entire host, for example, and so needing to have a way to co-exist with other software that may need to use ports 80 or 443.